usaspending_api/download/filestreaming/csv_generation.py

```diff
@@ -0,0 +1,284 @@
+            d2_filters = {'{}__isnull'.format(VALUE_MAPPINGS[award_level]['assistance_data']): False}
+            d2_source.queryset = queryset & award_level_table.objects.filter(**d2_filters)
+            csv_sources.append(d2_source)
+
+        verify_requested_columns_available(tuple(csv_sources), json_request.get('columns', []))
+
+    return csv_sources
+
+
+def parse_source(source, columns, download_job, working_dir, start_time, message, zipped_csvs, limit):
+    """Write to csv and zip files using the source data"""
+    d_map = {'d1': 'contracts', 'd2': 'assistance'}
+    source_name = '{}_{}'.format(d_map[source.file_type], VALUE_MAPPINGS[source.source_type]['download_name'])
+    source_query = source.row_emitter(columns)
+
+    reached_end = False
+    split_csv = 1
+    while not reached_end:
+        start_split_writing = time.time()
+        split_csv_name = '{}_{}.csv'.format(source_name, split_csv)
+        split_csv_path = os.path.join(working_dir, split_csv_name)
+
+        # Generate the query file; values, limits, dates fixed
+        temp_file, temp_file_path = generate_temp_query_file(split_csv, source_query, limit, source, download_job)
+
+        # Run the PSQL command as a separate Process
+        psql_process = multiprocessing.Process(target=execute_psql, args=(temp_file_path, split_csv_path,))
+        psql_process.start()
+
+        # Let the thread run until it finishes (max MAX_VISIBILITY_TIMEOUT), with a buffer of
+        # DOWNLOAD_VISIBILITY_TIMEOUT
+        while psql_process.is_alive() and (time.time() - start_time) < MAX_VISIBILITY_TIMEOUT:
+            if message:
+                message.change_visibility(VisibilityTimeout=DOWNLOAD_VISIBILITY_TIMEOUT)
+            time.sleep(60)
+
+        if (time.time() - start_time) >= MAX_VISIBILITY_TIMEOUT:
+            # Process is running for longer than MAX_VISIBILITY_TIMEOUT, kill it
+            write_to_log(message='Attempting to terminate process (pid {})'.format(psql_process.pid),
+                         download_job=download_job, is_error=True)
+            psql_process.terminate()
+
+            # Remove all temp files
+            os.close(temp_file)
+            os.remove(temp_file_path)
+            shutil.rmtree(working_dir)
+            zipped_csvs.close()
+            raise TimeoutError('Download job lasted longer than {} hours'.format(str(MAX_VISIBILITY_TIMEOUT/3600)))
+
+        # Write it to the zip
+        zipped_csvs.write(split_csv_path, split_csv_name)
+        write_to_log(message='Wrote {}, took {} seconds'.format(split_csv_name, time.time() - start_split_writing),
+                     download_job=download_job)
+
+        # Remove temporary files
+        os.close(temp_file)
+        os.remove(temp_file_path)
+
+        last_count = len(open(split_csv_path).readlines())
+        if last_count <= EXCEL_ROW_LIMIT:
+            # Will be hit when line 201 ((split_csv - 1) * EXCEL_ROW_LIMIT) > number of rows in source
```

Comment:
This is confusing (especially since it's inaccurate). Maybe avoid line numbers in comments?